File: 0069/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x <= 1:
            return x
        l, r = 1, x // 2
        res = 1
        while l <= r:
            p = (l + r) // 2
            square = p * p
            if square == x:
                return p
            if square < x:
                res = p
                l = p + 1
            else:
                r = p - 1
        return res

File: 0069/solution.ts
function mySqrt(x: number): number {
  if (x <= 1) {
    return x;
  }

  let res = 1,
    left = 0,
    right = Math.floor(x / 2);

  while (left <= right) {
    const pivot = Math.floor((left + right) / 2);
    const square = pivot * pivot;
    if (square === x) {
      return pivot;
    }
    if (square < x) {
      res = pivot;
      left = pivot + 1;
    } else {
      right = pivot - 1;
    }
  }
  return res;
}
